s3/replication/replicator/src/s3replicator/transfer_initiator.py

```diff
@@ -89,37 +91,94 @@ async def start(job, app):
             job.get_target_secret_key(),
             app_config.max_connections_per_s3_session)
 
-        if operation_type == ReplicationJobType.OBJECT_REPLICATION:
-            object_replicator = ObjectReplicator(
-                job, app["config"].transfer_chunk_size_bytes,
-                app["config"].range_read_offset,
-                app["config"].range_read_length,
-                source_session, target_session)
-            object_replicator.setup_observers(
-                "all_events", TranferEventHandler(app))
+        # For Cortx, default value of version_id=None
+        head_object = S3AsyncHeadObject(source_session, job.get_job_id(),
+                                        job.get_source_bucket_name(), job.get_source_object_name(),
+                                        None)
 
-            job.set_replicator(object_replicator)
-            job.mark_started()
+        await head_object.get(None)
 
-            # Start the replication.
-            semaphore = app['semaphore']
-            async with semaphore:
-                await object_replicator.start()
+        source_etag = head_object.get_etag()
+        _logger.info("Source ETag : {}".format(source_etag))
+
+        # List to store ETag of each parts of source object
+        part_length = []
+
+        # "-" in source ETag indicates multipart object upload
+        if "-" in source_etag:
+            # Get the total part count and object size
+            total_parts = source_etag.split("-")[1][0]
+            object_size = head_object.get_content_length()
+            _logger.debug("Total Parts : {}".format(total_parts))
+            _logger.debug("Content length : {}".format(object_size))
 
-        elif operation_type == ReplicationJobType.OBJECT_TAGS_REPLICATION:
-            object_tag_replicator = ObjectTagReplicator(
-                job, source_session, target_session)
-            object_tag_replicator.setup_observers(
-                "all_events", TranferEventHandler(app))
+            # calculate single part length
+            part_len = object_size / int(total_parts)
 
-            job.set_replicator(object_tag_replicator)
+            # Append the length of each part into the list
+            # Pass the list to multipart object replicator class
+            for part_no in range(1, int(total_parts) + 1):
+                _logger.debug("Part No : {}".format(part_no))
+                part_length.append(int(part_len))
```

Comment:
suppose we have an object, 100MB.
And every part is 15MB.
So, we will have 7 parts, and their sizes are:
part 1: 15MB
part 2: 15MB
part 3: 15MB
part 4: 15MB
part 5: 15MB
part 6: 15MB
part 7: 10MB

We are not dividing the objects into parts with equal sizes.
All parts except the last one have same sizes.
Only the last part could be less than standard size.

Reply:
Yes, this is correct.
Currently, Replicator is dividing the object by the part number as we are not getting specific part details. we are assuming that all parts are of fixed size. 
I have added few future references in the code. So, once we get the updated head object and get object api, we will modify the code and the replicator will only get specific parts from the source and put them in the destination.